src/components/Invoice/use-invoice.ts

```diff
@@ -0,0 +1,20 @@
+import { useQuery } from '@tanstack/react-query';
+
+const getInvoice =
+  (id: string) =>
+  ({ signal }: { signal: AbortSignal }) =>
+    fetch(`/api/invoices/${id}`, { signal })
+      .then((res) => {
+        if (res.status > 299) {
+          throw new Error(`Request failed with status ${res.status}`);
+        }
+
+        return res;
+      })
+      .then((res) => res.json());
```

Comment:
```ts
  .then(response => invoiceFormData.parse(response))
```

Reply:
no requests shown in inbox list with this

Reply:
fixed